File: ATM2/main.py
# Link to the problem: https://www.codechef.com/COOK98B/problems/ATM2

def main():
  T = int(input())
  while T:
    T -= 1
    N, K = list(map(int, input().split()))
    A = list(map(int, input().split()))
    ans = ''
    for i in range(N):
      if (K - A[i] >= 0):
        ans += '1'
        K -= A[i]
      else:
        ans += '0'
    print(ans)

if __name__ == '__main__':
  main()

File: ATM2/main.go
// Link to the problem: https://www.codechef.com/COOK98B/problems/ATM2

package main

import (
	"fmt"
)

func main() {
	var T int
	fmt.Scanf("%d", &T)
	for ; T > 0; T-- {
		var N, K int
		fmt.Scanf("%d%d", &N, &K)
		A := make([]int, N)
		for i := 0; i < N; i++ {
			fmt.Scanf("%d", &A[i])
		}
		Ans := ""
		for _, i := range A {
			if K-i >= 0 {
				Ans += "1"
				K -= i
			} else {
				Ans += "0"
			}
		}
		fmt.Println(Ans)
	}
}
